main.tf

```diff
@@ -184,14 +186,14 @@ resource "aws_cognito_identity_provider" "apple" {
 }
 
 resource "aws_cognito_user_pool_domain" "this" {
-  domain          = var.domain
+  domain          = var.domain                        # This value is mandatory and specifies the custom domain for the Cognito user pool.
```

Comment:
Domain now should be optional, if not defined, cognito must be using the auto-generated endpoint in the tf output